### PR TITLE
update participant to support signal broadcast skipping

### PR DESCRIPTION
**Pull request by paulwe**

### CODE REVIEW

File: pkg/rtc/participant.go [hunk 1]
```diff
@@ -1673,6 +1682,8 @@ func (p *ParticipantImpl) addMigrateMutedTrack(cid string, ti *livekit.TrackInfo
 }
 
 func (p *ParticipantImpl) addMediaTrack(signalCid string, sdpCid string, ti *livekit.TrackInfo) *MediaTrack {
+	p.requireBroadcast = true
```
Comment (davidzhao) [inline pkg/rtc/participant.go:1685]: I think this needs to be locked by `p.lock` ?

**Review by davidzhao — APPROVED** [review]

File: pkg/rtc/participant.go (hunk 1 above)
Comment (paulwe) [reply pkg/rtc/participant.go:1686]: yep, good catch

**Comment by boks1971** [comment]
Sorry, late to this. One question, when does `requireBroadcast` get reset. In the case of going from a publisher -> non-publisher, the flag gets set which is correct for the first broadcast after the change. Guessing we want to skip after that. Without having read the entire chain, wondering if it would make more sense to keep track of this in the `Room` object based on participant version (participant version right now already changes on read only when there is some underlying change based on the `dirty` flag which does get reset).

**Comment by davidzhao** [comment]
I think we cannot skip.. once a participant is decided to be broadcasted, we'll have to continue to broadcast. This is because otherwise it'll be "stuck" at a particular state.

**Comment by boks1971** [comment]
But, if there is no change, is it required? Wondering if the already existing change based (`dirty` flag) handling would prevent any unnecessary broadcast. I am surely missing some bit here. Will think more about it.

**Comment by davidzhao** [comment]
Right, if there is no change it wouldn't be needed.. I agree it requires some more thought.. but an element of it is that whether the participant is part of the broadcasted set or not would be a irreversible decision.

**Pull request merged**